docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,287 @@
+
+        - **`<plugin-name>`** *object*
+
+          This REQUIRED property specifies the plugin specification for plugin executable binary that validates the host and returns information about success or failure
+
+          - **`source`** *string*
+
+            This OPTIONAL property specifies an HTTP address of the plugin.
+
+          - **`data`** *object*
+
+            This OPTIONAL property specifies the input data for plugin executable binary. 
+
+    - **`refs`** *object*
+
+      This OPTIONAL property specifies the references to other namespaces.
+
+      - **`all`** *array of strings*
+
+        This OPTIONAL property specifies a list of all required namespaces that have to pass that container is considered to be compatible with a host.
+
+      - **`oneOf`** *array of strings*
+
+        This OPTIONAL property specifies a list of namespaces with a condition that at least one namespace has to pass that container is considered to be compatible with a host.
+
+      - **`noneOf`** *array of strings* 
+
+        This OPTIONAL property specifies a list of namespaces that cannot pass that container is considered to be compatible with a host.
+
+##### Example
+
+<details>
+  <summary>GPU pass-through</summary>
+
+```json
+{
+  "schema": "0.1.0",
+  "namespaces": {
+    "kernelMainlineV5": {
+      "plugins": {
+        "kernelVersion": {
+          "source": "<http-something-to-download-plugin>",
+          "data": {
+            "range": "5.0.0-5.15.0"
+          }
+        },
+        "refs": {
+          "all": [
+            "vfio",
+            "cpu",
+            "gpu"
+          ]
+        }
+      }
+    },
+    "vfio": {
+      "plugins": {
+        "kernelConfiguration": {
+          "data": {
+            "CONFIG_VFIO": "m",
+            "CONFIG_VFIO_IOMMU_TYPE1": "m",
+            "CONFIG_VFIO_MDEV": "m",
+            "CONFIG_VFIO_MDEV_DEVICE": "m",
+            "CONFIG_VFIO_PCI": "m"
+          }
+        },
+        "kernelModules": {
+          "data": {
+            "vfio": {},
+            "vfio_iommu_type1": {},
+            "vfio-pci": {}
+          }
+        }
+      }
+    },
+    "cpu": {
+      "refs": {
+        "oneOf": [
+          "cpuAmd",
+          "cpuIntel"
+        ]
+      }
+    },
+    "cpuAmd": {
+      "plugins": {
+        "hardware": {
+          "data": {
+            "cpu": ["AuthenticAMD"]
+          }
+        }
+      }
+    },
+    "cpuIntel": {
+      "plugins": {
+        "hardware": {
+          "data": {
+            "cpu": ["GenuineIntel"]
+          }
+        },
+        "kernelCmdline": {
+          "data": {
+            "intel_iommu": "on"
+          }
+        }
+      }
+    },
+    "gpu": {
+      "refs": {
+        "oneOf": [
+          "gpuNVIDIA",
+          "gpuIntel"
+        ]
+      }
+    },
+    "gpuNVIDIA": {
+      "plugins": {
+        "hardware": {
+          "data": {
+            "pci": ["0380:10de"]
```

Comment:
I would name it "vendor_id" pci could refer to to many id's

Reply:
the PCI field refers to PCI's `class_id:vendor_id:device_id`